src/ocl/batchnormocl.cpp

```diff
@@ -421,6 +421,76 @@ void BatchNormBackward(Handle& handle,
         MIOPEN_THROW(miopenStatusBadParm);
     }
 
+    const auto useSaved = savedMean != nullptr && savedInvVariance != nullptr;
+
+    const auto problem = batchnorm::ProblemDescription{
+        bn_mode, xDesc, dyDesc, dxDesc, bnScaleBiasDiffDesc, epsilon, useSaved};
+
+    const auto algo = bn_mode == miopenBNSpatial
+                          ? AlgorithmName{"miopenBatchNormBackwardPropSpatial"}
+                          : AlgorithmName{"miopenBatchNormBackwardPropPerActivation"};
+    const auto network_config = problem.MakeNetworkConfig();
+
+    const auto invoke_params = [&]() {
+        auto tmp              = batchnorm::BwdInvokeParams{};
+        tmp.type              = InvokeType::Run;
+        tmp.x                 = x;
+        tmp.dy                = dy;
+        tmp.dx                = dx;
+        tmp.bnScale           = bnScale;
+        tmp.resultBnScaleDiff = resultBnScaleDiff;
+        tmp.resultBnScaleDiff = resultBnScaleDiff;
+        tmp.resultBnBiasDiff  = resultBnBiasDiff;
+        tmp.epsilon           = epsilon;
+        tmp.savedMean         = savedMean;
+        tmp.savedInvVariance  = savedInvVariance;
+        return tmp;
+    }();
+
+    if(const auto existingInvoker = handle.GetInvoker(network_config, boost::none, algo))
+    {
+        (*existingInvoker)(handle, invoke_params);
+
+        if(miopen::CheckNumericsEnabled())
+        {
+            miopen::checkNumericsOutput(handle, dxDesc, dx);
+            miopen::checkNumericsOutput(handle, bnScaleBiasDiffDesc, resultBnScaleDiff);
+            miopen::checkNumericsOutput(handle, bnScaleBiasDiffDesc, resultBnBiasDiff);
+        }
+
+        return;
+    }
+    else
+    {
+        const auto ctx     = ExecutionContext{&handle};
+        const auto solvers = solver::SolverContainer<>{};
+        const auto slns    = solvers.SearchForSolutions(ctx, problem, 1);
+
+        // if(slns.empty())
+        //    MIOPEN_THROW(miopenStatusNotImplemented, "No solver found for activation forward.");
```

Comment:
?

Reply:
I will just uncomment this when I update all paths to solvers.

Reply:
Ok.